tests/integration-tests/tests/storage/kms_key_factory.py

```diff
@@ -0,0 +1,207 @@
+import json
+import logging
+import random
+import string
+import time
+
+import boto3
+import pkg_resources
+from jinja2 import Environment, FileSystemLoader
+
+
+class KMSKeyFactory:
+    """Manage creation for kms key."""
+
+    def __init__(self):
+        self.iam_client = None
+        self.kms_client = None
+        self.kms_key_id = None
+        self.account_id = boto3.client("sts").get_caller_identity().get("Account")
+        self.region = boto3.session.Session().region_name
+        self.iam_role = None
+        self.iam_policy_arn = None
+
+    def create_kms_key(self, scheduler):
+        """
+        Create a kms key given scheduler.
+        :param scheduler: Different scheduler need to attach different instance policies
+        """
+        # Only one iam role and one KMS key creation per factory allowed
+        if self.iam_role:
+            raise Exception("IAM Role already created")
+        if self.kms_key_id:
+            raise Exception("KMS key already created")
+
+        self.iam_role = self.create_role(scheduler)
+        self.kms_key_id = self._create_kms_key(scheduler)
+        return self.kms_key_id
+
+    def create_role(self, scheduler):
```

Comment:
Is this ever intended to be called on an instance of the object? Based on the fact that in this PR I only see it called in `create_kms_key`, I'm guessing not. If that's the case, can you rename this `_create_role` to be consistent with the naming convention used for the rest of the class?

Reply:
yes, done!